tests/test_builder.py

```diff
@@ -123,6 +109,107 @@ def test_build_runtime_insecure_registries(monkeypatch):
         )
 
 
+def test_build_runtime_target_image(monkeypatch):
+    get_k8s_helper_mock = unittest.mock.Mock()
+    monkeypatch.setattr(
+        mlrun.builder, "get_k8s_helper", lambda *args, **kwargs: get_k8s_helper_mock
+    )
+    mlrun.builder.get_k8s_helper().create_pod = unittest.mock.Mock(
+        side_effect=lambda pod: (pod, "some-namespace")
+    )
+    registry = "registry.hub.docker.com/username"
+    mlrun.mlconf.httpdb.builder.docker_registry = registry
+    mlrun.mlconf.httpdb.builder.function_target_image_name_prefix_template = (
+        "my-cool-prefix-{project}-{name}"
+    )
+    function = mlrun.new_function(
+        "some-function",
+        "some-project",
+        "some-tag",
+        image="mlrun/mlrun",
+        kind="job",
+        requirements=["some-package"],
+    )
+    image_name_prefix = mlrun.mlconf.httpdb.builder.function_target_image_name_prefix_template.format(
+        project=function.metadata.project, name=function.metadata.name
+    )
+
+    mlrun.builder.build_runtime(
+        mlrun.api.schemas.AuthInfo(), function,
+    )
+
+    # assert the default target image
+    target_image = (
+        mlrun.builder.get_k8s_helper()
+        .create_pod.call_args[0][0]
+        .pod.spec.containers[0]
+        .args[5]
+    )
+    assert target_image == f"{registry}/{image_name_prefix}:{function.metadata.tag}"
+
+    # assert we can override the target image as long as we stick to the prefix
+    function.spec.build.image = (
+        f"{registry}/{image_name_prefix}-some-addition:{function.metadata.tag}"
+    )
+    mlrun.builder.build_runtime(
+        mlrun.api.schemas.AuthInfo(), function,
+    )
+    target_image = (
+        mlrun.builder.get_k8s_helper()
+        .create_pod.call_args[0][0]
+        .pod.spec.containers[0]
+        .args[5]
```

Comment:
I think it is clearer what that "5" is, if you assign it to a variable and use it.
```suggestion
    container_image_index = 5
    target_image = (
        mlrun.builder.get_k8s_helper()
        .create_pod.call_args[0][0]
        .pod.spec.containers[0]
        .args[container_image_index]
```

Reply:
While I agree having "magic numbers" in the code (usually it happens in tests) is not a good practice, here I don't think moving it to a variable helps.
A good example when moving something to a variable helps can be found in `test_create_schedule_mlrun_function`
Instead of having the number 1 everywhere, we assigned it to the `expected_call_counter`, then we use it both to generate the trigger, and in the end to assert the number of runs triggered, the value here is both explainability, and also if someone want to change this test to check trigger of 2 runs, they'll just need to change one value.
Here we have several magic numbers (the zeros are magic numbers too), they all derived from the arg index in the relevant object (the index of the container in the conatiners list, the index of the call to the function, the index of the arg in the function etc...). Those numbers will need to be changed if someone for example changes the function signature, but I don't think there is a way to connect the two (so signature change won't break the tests)
Bottomline I don't think variable helps here (there's nothing to explain about this number, it's just the index of the container image argument), I will move the extraction of the target image from the mock to a function so if signature change we can change it one place